test/end2end_test.go

```diff
@@ -4636,6 +4636,33 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func (s) TestClientInitialHeaderEndStream(t *testing.T) {
```

Comment:
Could you please also add a test for a streaming RPC where the client does a close send (after headers) and then writes another data frame?

Reply:
Added that test (makes sure an RST is received in that case).